packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -63,21 +70,56 @@ test.describe('Chat Composite E2E Tests', () => {
     await browser.close();
   });
 
-  test('composite loads correctly with participant list', async () => {
-    await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].waitForSelector(dataUiId(IDS.participantList));
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('1-chat-screen.png');
+  test('composite pages load completely', async () => {
+    for (const idx in pages) {
+      await compositeLoaded(pages[idx]);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`1-page-${idx}-chat-screen.png`);
+    }
   });
 
   test('can send message', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
+    await page.keyboard.press('Enter');
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('2-send-message.png');
+  });
+
+  test('can receive message', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('3-receive-message.png');
+  });
+
+  test('read message has a viewed status', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="seen"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('4-read-message-status.png');
+  });
+
+  test('can view typing indicator', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'I am not superstitious. Just a little stitious.');
     await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
-    await pages[0].keyboard.press('Enter');
-    await pages[0].waitForSelector(`[data-ui-status="delivered"]`);
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('2-send-message.png');
+    await pages[0].waitForSelector(dataUiId(IDS.typingIndicator));
+    const el = await pages[0].$(dataUiId(IDS.typingIndicator));
+    expect(await el?.innerHTML()).toContain(participants[1]);
+    expect(await pages[0].screenshot()).toMatchSnapshot('5-typing-indicator.png');
+  });
+
+  test('typing indicator disappears after 12 seconds', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.waitForTimeout(12000);
```

Comment:
😢

Reply:
Is there anyway to mock timers and speed this up?

Reply:
Can't help it unfortunately.

Reply:
Hmm, I would say if we have to arbitrary wait 12 seconds we need to move this test to a non-playwright test were timers can be mocked

Given that typing indicator checks the Date.now() can we mock that https://stackoverflow.com/a/52876069?

Reply:
Vote to drop this test if there's no way to mock out the timers.

Reply:
Good idea @JamesBurnside 
Nan and I tried it earlier for mocking message timestamps but it didn't work.
Thankfully it worked for mocking typingindicator behavior.
Updated the code.

Note: we still have to wait a second.

Reply:
Unresolving conversation, still shows 12s - is there a local change waiting to be pushed?
If we still need the arbitrary wait for 1 second can you add a comment saying why (assume this is because our typing indicator has a `setTimeout((), 1000)` for checking if it should update the indicator)

Reply:
@JamesBurnside Probably forgot to push the changes. The code has now been updated.
Although typingindicator selector doesn't have a setTimeout, this wait is to ensure we don't encounter intermittent failures and allow selector to mutate the state based on the events received from underlying ACS SDK.